mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp

```diff
@@ -98,12 +99,23 @@ static DiagnosedSilenceableFailure unpackSingleIndexResultPayloadOperations(
       result.push_back(ofr);
       continue;
     }
-    auto payloadOps = state.getPayloadOps(ofr.get<Value>());
+
+    Value transformValue = ofr.get<Value>();
+    if (isa<ParamType>(transformValue.getType())) {
```

Comment:
When the type of `transformValue` is any type other than `ParamType` that implements `TransformParamTypeInterface`, this condition will fail and the code in line 113 will assert because there are no ops associated with param. This wasn't a problem before because the type constraint would only accept `ParamType` but it is relaxed in this commit. Please handle accordingly, e.g., by checking for `TransformParamTypeInterface` instead of hardcoded type here.